src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.cs

```diff
@@ -178,11 +213,47 @@ public override void Reset()
             Initialize();
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA5350", Justification = "HMACSHA1 is needed for compat. (https://github.com/dotnet/corefx/issues/9438)")]
+        private void OpenHmac(out HMAC hmac, out int hLen)
+        {
+            Debug.Assert(_password != null);
+
+            HashAlgorithmName hashAlgorithm = HashAlgorithm;
+            
+            if (string.IsNullOrEmpty(hashAlgorithm.Name))
+                throw new CryptographicException(SR.Cryptography_HashAlgorithmNameNullOrEmpty);
+
+            if (hashAlgorithm == HashAlgorithmName.SHA1)
+            {
+                hmac = new HMACSHA1(_password);
+                hLen = 160 >> 3;
```

Comment:
HMAC has two BlockSize values that I see: InputBlockSize and OutputBlockSize.  They're related to how it'll work under ICryptoTransform, so they're both 1.  But it does look like we have the `HashSize` property, which is the size in bits.  So _blockSize *can* be derived, simplifying this.